VMware.CloudFoundation.PowerManagement.psm1

```diff
@@ -1494,50 +1410,43 @@ Function Get-VMsWithPowerStatus {
+            if (-not $silence) { Write-PowerManagementLogMessage -Type INFO -Message "Connecting to '$server'..." }
             if ($DefaultVIServers) {
-                Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue  -ErrorAction  SilentlyContinue | Out-Null
+                Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Out-Null
             }
             Connect-VIServer -Server $server -Protocol https -User $user -Password $pass | Out-Null
             if ($DefaultVIServer.Name -eq $server) {
-                if(-not $silence) {Write-PowerManagementLogMessage -type INFO -Message "Connected to server '$server' and attempting to get the list of virtual machines..."}
-                    if ($pattern) {
-                        if ($PSBoundParameters.ContainsKey('exactMatch') ) {
-                            $no_of_vms = get-vm -Server $server | Where-Object Name -EQ $pattern  | Where-Object PowerState -eq $powerstate
-                        }
-                        else {
-                            $no_of_vms = get-vm -Server $server | Where-Object Name -match $pattern  | Where-Object PowerState -eq $powerstate
-                        }
-                    }
-                    else {
-                        $no_of_vms = get-vm -Server $server | Where-Object PowerState -eq $powerstate
-                    }
-                    if ($no_of_vms.count -eq 0) {
-                        if(-not $silence) {Write-PowerManagementLogMessage -type INFO -Message "No virtual machines in the $powerstate state."}
-                    }
-                    else {
-                        $no_of_vms_string = $no_of_vms -join ","
-                        if(-not $silence) {Write-PowerManagementLogMessage -type INFO -Message "The virtual machines in the $powerstate state are: $no_of_vms_string"}
+                if (-not $silence) { Write-PowerManagementLogMessage -Type INFO -Message "Connected to server '$server' and attempting to get the list of virtual machines..." }
+                if ($pattern) {
+                    if ($PSBoundParameters.ContainsKey('exactMatch') ) {
+                        $no_of_vms = get-vm -Server $server | Where-Object Name -EQ $pattern | Where-Object PowerState -EQ $powerstate
+                    } else {
+                        $no_of_vms = get-vm -Server $server | Where-Object Name -Match $pattern | Where-Object PowerState -EQ $powerstate
                     }
-                    Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue  -ErrorAction  SilentlyContinue | Out-Null
-                    Return $no_of_vms
-            }
-            else {
+                } else {
+                    $no_of_vms = get-vm -Server $server | Where-Object PowerState -EQ $powerstate
+                }
+                if ($no_of_vms.count -eq 0) {
```

Comment:
Or use Measure-Object.

Reply:
Could you please clarify this one?

Reply:
```powershell
if (-not $silence) {
    Write-PowerManagementLogMessage -Type INFO -Message "Connected to server '$server' and attempting to get the list of virtual machines..."
}

if ($pattern) {
    if ($PSBoundParameters.ContainsKey('exactMatch')) {
        $virtualMachines = Get-VM -Server $server | Where-Object { $_.Name -eq $pattern -and $_.PowerState -eq $powerstate }
    } else {
        $virtualMachines = Get-VM -Server $server | Where-Object { $_.Name -match $pattern -and $_.PowerState -eq $powerstate }
    }
} else {
    $virtualMachines = Get-VM -Server $server | Where-Object { $_.PowerState -eq $powerstate }
}

$vmCount = ($virtualMachines | Measure-Object).Count

if ($vmCount -eq 0) {
    if (-not $silence) {
        Write-PowerManagementLogMessage -Type INFO -Message "No virtual machines in the $powerstate state."
    }
} else {
    $virtualMachinesList = $virtualMachines -join ","
    if (-not $silence) {
        Write-PowerManagementLogMessage -Type INFO -Message "The virtual machines in the $powerstate state are: $virtualMachinesList"
    }
}
```

Reply:
The code looks nice and easy to read but will not change it and in this way will reduce the testing scope.